### PR TITLE
Revert "util: Flush caches after notifying for memhooks and rocr"

### PR DESCRIPTION
**Pull request by wckzhang**
Calling malloc (which happens in rdma-core during deregistration if fork
support is enabled) causes a deadlock if called in the sbrk memhook.
glibc acquires a mutex in _int_free that is also acquired in
__libc_malloc.

See #6244

This reverts commit d10df326973edcf2bf00cccaf3bf90168636fafb.

### CODE REVIEW

**Comment by wckzhang** [comment]
@shefty let me know if you want me to revert any of the other patches in the PR series. I figured we can just leave the windows rwlock implementation in, but there's no need to use it anymore or have the cache flush function. This revert should restore correctness for us though.

**Comment by shefty** [comment]
I would go with the minimal revert like you did

File: prov/util/src/util_mem_monitor.c [hunk 1]
```diff
@@ -347,15 +347,7 @@ void ofi_monitor_unsubscribe(struct ofi_mem_monitor *monitor,
 #include <sys/ioctl.h>
 #include <linux/userfaultfd.h>
 
-/* The userfault fd monitor requires for events that could
```
Comment (rwespetal) [inline prov/util/src/util_mem_monitor.c:350]: This comment is probably worth leaving in for future reference.

**Review by rwespetal — APPROVED** [review]

File: prov/util/src/util_mem_monitor.c (hunk 1 above)
Comment (wckzhang) [reply prov/util/src/util_mem_monitor.c:350]: Let me split that off into its own commit

**Review by shefty — APPROVED** [review]

**Comment by wckzhang** [comment]
I don't think that travis failure is related to this patch

**Pull request merged**